v3.0.0: Удаление записи полей с параметрами приложения в window

kurakinasv opened the pull request:
**Задача [SPECIAL-11631](https://youtrack.kts.studio/issue/SPECIAL-11631/mediaproject-vk-Ubrat-zapis-v-window-polej-s-parametrami-prilozheniya-dlya-miniappa-VK)**:

- отрефакторить утилиту `checkVkPlatform`, чтобы она не обращалась к window
- убрать лишние поля из типа `WindowType`

**Что было сделано:**

- Опираясь на [пример в другом проекте](https://git.kts.tech/mailru-nonstandart/sber-chan-comics-front/-/blob/main/src/store/globals/appParams/AppParamsStore/AppParamsStore.ts), отрефакторены утилиты `checkVkPlatform`, `setVkViewSettings` и `initializeVkApp` – удалены обращения к window
- Из утилиты `getNewVkAccessToken` убрано значение по умолчанию. В соответствии с этим немного подкорректированы другие методы, которые так или иначе связаны с запросом за токеном
- Из `WindowType` убраны поля, которые не должны храниться в window: параметры запуска и параметры определения платформы
- Подняла версию до 3.0.0, так как по крайней мере в функции `setVkViewSettings` было изменено количество аргументов

**Как тестировалось:**

1. Закомменчена часть с авторизацией в проекте
2. Из mediaproject-utils и из mediaproject-stores выполнить: `yarn link && yarn build`. Используемая [версия stores](https://github.com/ktsstudio/mediaproject-stores/pull/11), и [версия utils](https://github.com/ktsstudio/mediaproject-utils/pull/16)
3. Из mediaproject-vk выполнить: `yarn build` 
4. И затем выполнить то, что описано [здесь](https://github.com/ktsstudio/mediaproject-vk/pull/15#:~:text=%D0%9A%D0%B0%D0%BA%20%D0%BB%D0%BE%D0%BA%D0%B0%D0%BB%D1%8C%D0%BD%D0%BE%20%D0%BF%D1%80%D0%BE%D1%82%D0%B5%D1%81%D1%82%D0%B8%D1%80%D0%BE%D0%B2%D0%B0%D1%82%D1%8C%20%D0%B1%D0%B8%D0%B1%D0%BB%D0%B8%D0%BE%D1%82%D0%B5%D0%BA%D1%83), начиная с пункта iii: скопировать папку dist и package.json в другой каталог и выполнить из него `yarn install && npm link <путь до проекта>/node_modules/react && yarn link`
5. В проекте выполнить: `yarn link @ktsstudio/mediaproject-utils && yarn link @ktsstudio/mediaproject-stores && yarn link @ktsstudio/mediaproject-vk && yarn dev`
6. Заменить использование локальных функций `checkVkPlatform`, `setVkViewSettings` и типов `DeviceInfo` и `VkPlatformType` на одноименные из либы
7. Заменить другие утилиты на одноименные из mediaproject-utils и -stores (описано в пр'ах по ссылкам выше)
8. И, например, проверить, инициализируются ли в сторе VkAppParamsStore параметры приложения

**Чтобы все отвязать:**

1. В проекте выполнить: `yarn unlink @ktsstudio/mediaproject-vk && yarn unlink @ktsstudio/mediaproject-utils && yarn unlink @ktsstudio/mediaproject-stores && yarn install --force`
2. В каталоге с собранной библиотекой: `npm uninstall -g react && yarn unlink`
3. В сторах и утилзах: `yarn unlink`

Commit:
[*] access to window object; WindowType fields

Commit:
[*] removed default appId argument

Commit:
[*] version, changelog

AliceSitedge commented on src/callVkApi.ts:
```diff
@@ -39,7 +39,7 @@ const callVkApi = async <D = any>({
   accessToken = null,
   renewTokenIfExpired = true,
-  getAccessTokenParams = {},
+  getAccessTokenParams,
```
С текущими изменениями у функции меняется поведение. Раньше, если в getAccessTokenParams ничего не передано, запрос за токеном все равно происходил, потому что appId функция могла достать самостоятельно. А сейчас получается запроса за токеном не будет и запрос к апишке будет неуспешный в случае отсутствия токена

Как будто лучше getAccessTokenParams сделать обязательным аргументом, чтобы appId всегда передавался

Commit:
[*] pr fixes

- getAccessTokenParams is required in callVkApi
- checkVkPlatform: void -> null
- setVkViewSettings: arguments order

Commit:
[-] initializeVkApp

Commit:
[*] js doc

Commit:
[*] fix

dewolixgit commented:
Переназначил целевую ветку с master на release-3.0.0 для подготовки PR по релизу 3.0.0
